extensions/spring-cloud-config-client/runtime/src/main/java/io/quarkus/spring/cloud/config/client/runtime/VertxSpringCloudConfigGateway.java

```diff
@@ -77,15 +77,16 @@ public static WebClient createHttpClient(Vertx vertx, SpringCloudConfigClientCon
                 }
             } else if (trustAll) {
                 skipVerify(webClientOptions);
-            } else if (springCloudConfig.keyStore.isPresent()) {
-                Path trustStorePath = springCloudConfig.keyStore.get();
-                String type = determineStoreType(trustStorePath);
-                KeyStoreOptionsBase storeOptions = storeOptions(trustStorePath, springCloudConfig.keyStorePassword,
+            }
+            if (springCloudConfig.keyStore.isPresent()) {
+                Path keyStorePath = springCloudConfig.keyStore.get();
+                String type = determineStoreType(keyStorePath);
+                KeyStoreOptionsBase storeOptions = storeOptions(keyStorePath, springCloudConfig.keyStorePassword,
                         createStoreOptions(type));
                 if (isPfx(type)) {
-                    webClientOptions.setPfxTrustOptions((PfxOptions) storeOptions);
+                    webClientOptions.setPfxKeyCertOptions((PfxOptions) storeOptions);
                 } else {
-                    webClientOptions.setTrustStoreOptions((JksOptions) storeOptions);
+                    webClientOptions.setKeyStoreOptions((JksOptions) storeOptions);
```

Comment:
Could you explain the change of method? Does it still work in the initial case covered?

Reply:
Do you mean SSL session without checking client cert on a server? Usual way in Java is using truststore for this purpose. In that case keystore not needed. From my point of view using keystore as a store for CA certificate is a mistake.

For this change I was guided by https://vertx.io/docs/vertx-core/java/#ssl

Reply:
@michalszynkiewicz you probably have more experience than me on the Vert.x web client, could you have a look?

Reply:
This looks good to me.
Setting trust store options from keystore didn't look good.